src/ControllerV2/BeneficiaryPersonalDataController.php

```diff
@@ -0,0 +1,452 @@
+            ])->getContent(),
+        ]);
+    }
+
+    #[Route(
+        path: '/{id}/folder/{folderId}/search',
+        name: 'folder_search',
+        requirements: ['id' => '\d+', 'folderId' => '\d+'],
+        methods: ['POST'],
+        condition: 'request.isXmlHttpRequest()',
+    )]
+    #[ParamConverter('folder', class: 'App\Entity\Dossier', options: ['id' => 'folderId'])]
+    #[IsGranted('UPDATE', 'beneficiary')]
+    public function searchFolders(
+        Request $request,
+        Beneficiaire $beneficiary,
+        Dossier $folder,
+        FolderableItemManager $manager,
+        PaginatorService $paginator,
+    ): JsonResponse {
+        $this->denyAccessUnlessGranted('UPDATE', $folder);
+
+        $searchForm = $this->getSearchForm(
+            $this->generateUrl('folder_search', ['id' => $beneficiary->getId(), 'folderId' => $folder->getId()]),
+        )->handleRequest($request);
+
+        $search = $searchForm->get('search')->getData();
+
+        return new JsonResponse([
+            'html' => $this->renderForm('v2/vault/document/_list.html.twig', [
+                'foldersAndDocuments' => $paginator->create(
+                    $manager->getFoldersAndDocumentsWithUrl($beneficiary, $folder, $search),
+                    $request->query->getInt('page', 1),
+                ),
+                'beneficiary' => $beneficiary,
+                'form' => $searchForm,
+            ])->getContent(),
+        ]);
+    }
+
+    #[Route(
+        path: '/{id}/folder/create',
+        name: 'folder_create',
+        requirements: ['id' => '\d+'],
+        methods: ['GET', 'POST'],
+    )]
+    #[IsGranted('UPDATE', 'beneficiary')]
+    public function createFolder(
+        Request $request,
+        Beneficiaire $beneficiary,
+        EntityManagerInterface $em,
+        FolderManager $manager
+    ): Response {
+        $folder = (new Dossier())->setBeneficiaire($beneficiary);
+        $form = $this->getCreateForm(
+            FolderType::class,
+            $folder,
+            $this->generateUrl('folder_create', ['id' => $beneficiary->getId()]),
+        )->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $em->persist($folder);
+            $em->flush();
+
+            return $this->redirectToRoute('document_list', ['id' => $beneficiary->getId()]);
+        }
+
+        return $this->renderForm('v2/vault/folder/create.html.twig', [
+            'form' => $form,
+            'beneficiary' => $beneficiary,
+            'autocompleteNames' => $manager->getAutocompleteFolderNames(),
+        ]);
+    }
+
+    private function getSearchForm(string $url): FormInterface
+    {
+        return $this->createForm(SearchType::class, null, [
+            'attr' => ['data-controller' => 'ajax-list-filter'],
+            'action' => $url,
+        ]);
+    }
+
+    private function getCreateForm(string $type, DonneePersonnelle $entity, string $url): FormInterface
```

Comment:
$type is not a good variable name I think, maybee $formType, but still, it does not represent what it is, it should be more `$formTypeClassName`, or `$formFQCN`